hybris.c

```diff
@@ -1527,6 +1608,170 @@ led_control_hammerhead_probe(led_control_t *self)
+  if( state->blink )
+    dprintf(state->fd_ledreset, "%d", 0);
+
+  (state+0)->brightness = led_util_scale_value(r, (state+0)->maxval);
+  (state+1)->brightness = led_util_scale_value(g, (state+1)->maxval);
+  (state+2)->brightness = led_util_scale_value(b, (state+2)->maxval);
+
+  dprintf((state+0)->fd_brightness, "%d", (state+0)->brightness);
+  dprintf((state+1)->fd_brightness, "%d", (state+1)->brightness);
+  dprintf((state+2)->fd_brightness, "%d", (state+2)->brightness);
+
+  if( state->blink ) {
+    // need to reset the blink when changing color (do we?)
+    dprintf(state->fd_grpfreq, "%d", state->freq); // 1s
+    dprintf(state->fd_grppwm, "%d", state->pwm); // 50%?
+    dprintf(state->fd_blink, "%d", 1);
+  } else
+    dprintf(state->fd_blink, "%d", 0);
+}
+
+static void
+led_control_bacon_close_cb(void *data)
+{
+  led_state_bacon_t *state = data;
+  led_state_bacon_close(state + 0);
+  led_state_bacon_close(state + 1);
+  led_state_bacon_close(state + 2);
+}
+
+static bool
+led_control_bacon_probe(led_control_t *self)
+{
+  /** Sysfs control paths for RGB leds */
+  static const led_paths_bacon_t paths[][3] =
+  {
+    // bacon
+    {
+      {
+        .brightness = "/sys/class/leds/red/brightness",
+        .grpfreq    = "/sys/class/leds/red/device/grpfreq",
+        .grppwm     = "/sys/class/leds/red/device/grppwm",
+        .blink      = "/sys/class/leds/red/device/blink",
+        .ledreset   = "/sys/class/leds/red/device/ledreset",
+      },
+      {
+        .brightness = "/sys/class/leds/green/brightness",
+        .grpfreq    = "/sys/class/leds/green/device/grpfreq",
+        .grppwm     = "/sys/class/leds/green/device/grppwm",
+        .blink      = "/sys/class/leds/green/device/blink",
+        .ledreset   = "/sys/class/leds/green/device/ledreset",
+      },
+      {
+        .brightness = "/sys/class/leds/blue/brightness",
+        .grpfreq    = "/sys/class/leds/blue/device/grpfreq",
+        .grppwm     = "/sys/class/leds/blue/device/grppwm",
+        .blink      = "/sys/class/leds/blue/device/blink",
+        .ledreset   = "/sys/class/leds/blue/device/ledreset",
+      }
+    },
+  };
+
+  static led_state_bacon_t state[3];
+
+  bool res = false;
+
+  led_state_bacon_init(state+0);
+  led_state_bacon_init(state+1);
+  led_state_bacon_init(state+2);
+
+  self->name   = "bacon";
+  self->data   = state;
+  self->enable = led_control_bacon_enable_cb;
+  self->blink  = led_control_bacon_blink_cb;
+  self->value  = led_control_bacon_value_cb;
+  self->close  = led_control_bacon_close_cb;
+
+  // need to check 
+  self->can_breathe = false;
+
+  mce_log(LOG_ERR, "bacon probe");
+  for( size_t i = 0; i < G_N_ELEMENTS(paths) ; ++i )
+  {
+    if( led_state_bacon_probe(&state[0], &paths[i][0]) &&
+        led_state_bacon_probe(&state[1], &paths[i][1]) &&
+        led_state_bacon_probe(&state[2], &paths[i][2]) )
+    {
+      mce_log(LOG_ERR, "bacon probed!");
```

Comment:
Use INFO/DEBUG as this is not an error situation.

nb During debugging you can run mce manually, something like

    mce -qqq -T -lhybris.c:*

to get verbose logging just from this plugin.